reference_guide/navigation/navigation_intro.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Introduction to Navigation
+---
+
+Objects that aggregate the 
+[Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface can be displayed in the IDE. The role of the `Navigatable` interface is to facilitate opening an object representing content
+in the appropriate editing or display window, and requesting focus. For example:
+* Opening [Files, classes or symbols](https://www.jetbrains.com/help/idea/navigating-to-class-file-or-symbol-by-name.html),
+  in an editor. This also applies to editing content in preference windows, such as 
+  [Copyright profiles](https://www.jetbrains.com/help/idea/copyright-profiles.html).
+* Going to locations within files based on [Bookmarks](https://www.jetbrains.com/help/idea/navigating-with-bookmarks.html),
+  [TODOs](https://www.jetbrains.com/help/idea/using-todo.html), 
+  or [Lines](https://www.jetbrains.com/help/idea/2017.3/navigating-to-line.html). 
+
+### The Navigatable Interface 
+The [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface defines the following methods of particular interest:
+
+```java
+
+ boolean canNavigate();
+
+ boolean canNavigateToSource();
+ 
+ void navigate(boolean requestFocus); 
+```
+The `canNavigate()` method indicates whether this object supports navigation of any kind. The `canNavigateToSource()` method
+indicates whether this object supports navigation to source within an editor. See the 
+[Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface comments for more information about these two methods.
+
+The `navigate()` method is called to open the (content-containing) object in an editor or display, and request focus. Typically an implementation
+of `Navigatable` does the heavy lifting of invoking an editor for the content and requesting focus. See below for examples,
+and the [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface for more information.  
+
+### Example IntelliJ Platform Implementations of Navigatable
+This section describes some example implementations that can be found in the IntelliJ Platform. This is by no means an exhaustive list, 
+but is intended to illustrate how `Navigatable` works in different contexts.
```

Comment:
`NavigatablePsiElement` and `OpenFileDescriptor` are not "example implementations", these are the main two ways plugin developers will interact with navigation functionality. This particular section should have a title "Navigating to PSI Elements" and should say that:
  * most PSI elements support navigation by implementing `NavigatablePsiElement`
  * if you have a PSI element and want to navigate to it, you can usually do that by casting it to `NavigatablePsiElement` and then invoking `navigate()`
  * if you implement a PSI element subclass, you usually want to support navigation for it, and you can do this by either ensuring that your base class implements `NavigatablePsiElement` or, if not, by implementing it yourself.